system/jlib/jdebug.cpp

```diff
@@ -364,7 +364,9 @@ cycle_t jlib_decl get_cycles_now()
     if (useRDTSC)
         return getTSC();
 #endif
-#ifndef __APPLE__
+#ifdef __APPLE__
+    return mach_absolute_time() * (uint64_t) timebase_info.numer / (uint64_t)timebase_info.denom;
```

Comment:
Maybe, though it's all a bit moot with changes you are proposing to get_cycles_now - it will only hit this code if rdtsc is not available.